packages/astro/src/core/session.ts

```diff
@@ -138,9 +152,10 @@ export class AstroSession<TDriver extends SessionDriverName = any> {
 				message: 'The session key was not provided.',
 			});
 		}
+		let cloned: T;
 		try {
 			// Attempt to serialize the value so we can throw an error early if needed
```

Comment:
The comment should be updated to reflect the new logic

Reply:
Agreed. Can you include the explanation of why we're cloning it too.